src/zap/zapreadytorun.cpp

```diff
@@ -175,16 +175,38 @@ class EntryPointVertex : public NativeFormat::Vertex
     }
 };
 
+class EntryPointWithBlobVertex : public EntryPointVertex
+{
+    BlobVertex * m_pBlob;
+
+public:
+    EntryPointWithBlobVertex(DWORD methodIndex, BlobVertex * pFixups, BlobVertex * pBlob)
+        : EntryPointVertex(methodIndex, pFixups), m_pBlob(pBlob)
+    {
+    }
+
+    virtual void Save(NativeWriter * pWriter)
+    {
+        pWriter->WriteUnsigned(m_pBlob->GetSize());
```

Comment:
It should not be necessary to encode the size - we do not do it anywhere else for the native signatures.